content/id/docs/concepts/extend-kubernetes/api-extension/custom-resources.md

```diff
@@ -0,0 +1,221 @@
+
+CRD memungkinkan pengguna untuk membuat tipe baru *resource* tanpa menambahkan APIserver lain. Kamu tidak perlu mengerti Agregasi API untuk menggunakan CRD.
+
+Terlepas dari bagaimana cara mereka dipasang, *resource* baru disebut sebagai Custom Resources untuk memisahkan mereka dari *resource* bawaan Kubernetes (seperti Pod).
+
+## CustomResourceDefinitions
+
+Sumber daya API [CustomResourceDefinition](/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/) memungkinkan kamu untuk medefinisikan Custom Resources. Mendefinisikan sebuah objek CRD akan membuat sebuah Custom Resource dengan sebuah nama dan skema yang kamu spesifikasikan. Kubernetes API melayani dan menangani penyimpanan dari Custom Resource kamu.
+
+Ini membebaskan kamu dari menulis server API kamu sendiri untuk menangani Custom Resource, tetapi sifat dasar dari implementasi menyebabkan kamu memiliki fleksibilitas yang berkurang dibanding [agregasi server API](#api-server-aggregation).
+
+Lihat [contoh *custom controller*](https://github.com/kubernetes/sample-controller) sebagai sebuah contoh dari bagaimana cara untuk mendaftarkan sebuah Custom Resource, bekerja dengan instans dari tipe baru *resource* kamu, dan menggunakan sebuah *controller* untuk menangani *event*.
+
+## Agregasi server API
+
+Biasanya, tiap *resource* di API Kubernetes membutuhkan kode yang menangani permintaan REST dan mengatur peyimpanan tetap dari objek-objek. Server Kubernetes API utama menangani *resource* bawaan seperti Pod dan Service, dan juga menangani Custom Resource dalam sebuah cara yang umum melalui [CRD](#customresourcedefinitions).
+
+[Lapisan agregasi](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/) memungkinkan kamu untuk menyediakan implementasi khusus untuk Custom Resource dengan menulis dan men-deploy API server kamu yang berdiri sendiri. API server utama menlimpahkan permintaan kepada kamu untuk Custom Resource yang kamu tangani, membuat mereka tersedia untuk semua kliennya.
+
+## Memilih sebuah metode untuk menambahkan Custom Resource
+
+CRD lebih mudah digunakan. Aggregated API lebih fleksibel. Pilih metode yang paling baik untuk kebutuhan kamu.
+
+Biasanya, CRD cocok jika:
+
+* Kamu memiliki *field* yang banyak
+* Kamu menggunakan *resource* dalam perusahaan kamu, atau sebagau bagian dari proyek *open-source* kecil (berlawanan dengan sebuah produk komersil)
+
+### Membandingkan kemudahan penggunaan
+
+CRD lebih mudah dibuat dibandingkan dengan Aggregated API.
+
+| CRD                        | API Teragregasi |
+| --------------------------- | -------------- |
+| Tidak membutuhkan pemrograman. Pengguna dapat memilih bahasa apapun untuk sebuah *controller* CRD. | Membutuhkan pemrograman dalam Go dan membangun *binary* dan *image*. Pengguna dapat memilih bahasa apapun untuk sebuah CRD *controller*. |
+| Tidak ada servis tambahan yang dijalankan; CRs ditangani oleh server API. | Sebuah servis tambahan untuk menciptakan dan dapat gagal. |
```

Comment:
we should eliminate plural for english: just "CR" or "CR-CR"
```suggestion
| Tidak ada servis tambahan yang dijalankan; CR ditangani oleh server API. | Sebuah servis tambahan untuk menciptakan dan dapat gagal. |
```

Reply:
I'll go with CR